### PR TITLE
Add mock example for sub-query

### PR DESCRIPTION
**Pull request by janeksledziewski**

### CODE REVIEW

File: website/docs/advanced-usage/mocking.md [hunk 1]
```diff
@@ -101,3 +101,28 @@ private class ExampleControllerTest {
...
+
+### Sub-Query
+
+To mock a sub-query create a JSON String with expected data and deserialize it to expected SObjectType. Then pass this data do SOQL class using `.setMock` method.
+
+```
+@IsTest
+static void checkMocking() {
+    List<Account> mocks = (List<Account>) JSON.deserialize(
+        '[{ "Name": "Account Name", "Contacts": { "totalSize": 1, "done": true, "records": [{ "Name": "Contact Name", "Email": "contact.email@address.com" }] }  }],
```
Comment (pgajek2) [inline website/docs/advanced-usage/mocking.md:113]: Maybe:
```java
List<Account> mocks = (List<Account>) JSON.deserialize(JSON.serialize(new Map<String, Object>{
    'Name' => 'Account Name',
    'Contacts => new Map<String, Object>{
        'totalSize' => 1, 
        'done' => true, 
        'records' => new List<Contact>{
             new Contact(
                  Name = 'Contact Name',
                  Email = 'contact.email@address.com'
            )
        }
    }
}) ,List<Account>.class);
```

File: website/docs/examples/mocking.md [hunk 2]
```diff
@@ -175,3 +175,81 @@ public class ExampleControllerTest {
...
+}
+```
+
+Deserialize desired data from JSON format to selected SObjectType. And pass data in form of single record or list of records.
+
+```
+@IsTest
+static void getAccountsWithContacts() {
+    List<Account> mocks = (List<Account>) JSON.deserialize(
+        '[{ "Name": "Account Name", "Contacts": { "totalSize": 1, "done": true, "records": [{ "Name": "Contact Name", "Email": "contact.email@address.com" }] }  }],
```
Comment (pgajek2) [inline website/docs/examples/mocking.md:204]: Może tutaj też Mapa?

Comment (janeksledziewski) [reply website/docs/examples/mocking.md:204]: It provides additional options, below is an example with Map

Comment (pgajek2) [reply website/docs/examples/mocking.md:204]: Sounds good!

**Review by pgajek2 — APPROVED** [review]

**Pull request merged**